File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::map<int, int> holder;
        for (int i = 0; i < nums.size(); i++) {
            if (holder[nums[i]] == 1) {
                return true;
            }
            else {
                holder[nums[i]] = 1;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/house-robber/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        ans = 0
        n = len(nums)
        dp = [0] * n
        for i, num in enumerate(nums):
            if i == 0:
                dp[0] = num
            elif i == 1:
                dp[i] = max(num, dp[i-1]) # rob either last house or this house
            else:
                dp[i] = max(dp[i-1], num + dp[i-2]) # rob either this house and get the max you can from the previous houses, or don't rob
        return dp[-1]


            